dev/tests/proxy.rs

```diff
@@ -220,6 +224,29 @@ async fn simple_backend_proxy() {
         .await
         .unwrap();
 
+    let hostname = format!("{}.{}", "foobar", CLUSTER);
+    let client = ClientBuilder::new()
+        .redirect(Policy::none())
+        .resolve(&hostname, proxy.bind_redir_address)
+        .build()
+        .unwrap();
+
+    let req = client.get(format!(
+        "http://{}:{}{}",
+        hostname,
+        proxy.bind_redir_address.port(),
+        "/"
+    ));
+    let response = req.send().await.unwrap();
+
+    assert_eq!(StatusCode::PERMANENT_REDIRECT, response.status());
+    let mut https_uri = response.url().clone();
+    https_uri.set_scheme("https").unwrap();
+    assert_eq!(
+        response.headers().get("LOCATION").unwrap(),
+        https_uri.as_str()
+    );
+
```

Comment:
Should we have an assertion that makes sure the path matches too?

Reply:
I'm checking the whole URL, so that includes the path.

Reply:
but I think I'm misunderstanding you.

Reply:
I see what you mean. You could throw `/foo/bar` into the path of the request so that an implementation that failed to preserve the path would fail.